### PR TITLE
Updated instructions for the frontmatter

### PR DESCRIPTION
**Pull request by filbabic**
Hey there @jellodiil!

I added a bunch of instructions to the frontmatter files with some examples of how to fill it in.

Let me know if there's anything to fix here!

I'll add Ray to the PR too, so he can see it once he gets back from vacay.

Thanks!

### CODE REVIEW

**Comment by sammyd** [comment]
I would strongly recommend against putting extensive sample text in this template. It is significantly more difficult for somebody reviewing the book to realise that a section of text has been erroneously left in / not updated to match the correct book than it is to search for TODOs.

I would suggest that keeping all TODOs as a single line, prefixed with TODO. If a sample is required, link to one in a different repo, or a published book on the site. This makes it much more likely that if a book is TODO-free then it is OK to publish.

**Comment by filbabic** [comment]
@sammyd The reason I added longer sample texts in is because people just don't know what to write for certain parts of the book.

Additionally, some parts of the book don't even show up on the online reader (as we discussed) and I can't just link to them in this repo. Not everyone on our Book team has access to all the books, they can only access the books they're working on. (Unless I'm mistaken)

- It is significantly more difficult for somebody reviewing the book to realise that a section of text has been erroneously left in / not updated to match the correct book than it is to search for TODOs.

In this case, we can add a "Remove This" text after each "Example" heading. Does that work?

- If a sample is required, link to one in a different repo, or a published book on the site. This makes it much more likely that if a book is TODO-free then it is OK to publish.

I can add a part in the guide where it talks about cleaning up ALL the TODOs from the frontmatter etc before publishing the book. Does that work?

Again, we can't just link to examples as some parts don't show up online but they're important.

Also, we're trying to add these examples and instructions here, so that we only have to change it in one place. This is why we don't want these instructions in the guide, as it would be more overhead having to fill them in multiple places.

Let me know what you think!

**Comment by sammyd** [comment]
Ah—I take your point that it's not possible to link to other places yet. I hadn't considered that.

An alternative could be to create a `samples` directory within this repo that the TODOs refer to. Or to store the samples in the guides, and refer to them in the TODOs.

But maybe I'm imagining a problem that doesn't really exist. I expect that Edith will be in a better position to state whether this is likely to be a problem or not. I'd take what she says over my minimal experience any day.

Either way, it's pretty easy to change in the future.

**Comment by filbabic** [comment]
@sammyd Yeah makes sense.

The `samples` directory` is a good idea, as that way it'd be easier to directly update the repos. It's also hard to add samples to the guides, because then the guides are bloated and have 400-word-examples of the Intro, Conclusion, or What You Need pages.

I'll wait for Edith's input here, and see if we can come to a mutually pain-free approach!

Thanks!

File: frontmatter/cover.markdown [hunk 1]
```diff
@@ -1 +1,3 @@
+[TODO: Book Team Lead - Add cover]
+
 ![cover](images/cover.png)
```
Comment (jellodiil) [inline frontmatter/cover.markdown]: This file never needs changing. the cover is always called cover, the TODO should be in publish.yaml to check the newest one is uploaded. I do that.

File: frontmatter/what-you-need.markdown [hunk 2]
```diff
@@ -13,3 +26,11 @@ To follow along with this book, you'll need the following:
 
 * **Technology Two**: Fill out what the reader needs; a certain OS version, a certain IDE version, etc.
 
+# Example
```
Comment (jellodiil) [inline frontmatter/what-you-need.markdown]: I don't think an example is needed here. It is quite obvious from the above section what is needed, I don't think an example will provide anymore clarity.

Comment (filbabic) [reply frontmatter/what-you-need.markdown]: Hey Edith!

You'd think that, but then you take a look at this book for example: https://www.raywenderlich.com/books/dagger-by-tutorials/v1.0/chapters/i-what-you-need and there is no language and no Android or platform version, or Dagger/Hilt versions included.

One of the reasons we have all these instructions is that even some things which are basic logic are often not filled. :shrug:

File: frontmatter/cover.markdown (hunk 1 above)
Comment (filbabic) [reply frontmatter/cover.markdown]: I'll update it & remove the instructions.

I can add a TODO to yaml if you want!

File: frontmatter/what-you-need.markdown (hunk 2 above)
Comment (jellodiil) [reply frontmatter/what-you-need.markdown]: Yes, but leaving an in-depth example doesn't mean that people will add any more information. Could be that these were not important to the author a the time of writing, and the minimum needed is just what they added there? or maybe it is addressed somehow in the book that makes it obvious not to add here. 
What I am saying is, you've expanded the "what you need" section already by adding lines for what we'd expect like the Framework and OS versioning which already have samples in them, I just feel that adding a specific example is too much, and not necessarily helpful.

**Comment by jellodiil** [comment]
*  >I would strongly recommend against putting extensive sample text in this template. It is significantly more difficult for somebody reviewing the book to realise that a section of text has been erroneously left in / not updated to match the correct book than it is to search for TODOs.

I definitely agree with this. Even from an example point of view, a fully Android example from another book might be irrelevant and confusing to a FPE who has only ever written in iOS. And secondly, as Sam said it is then hard to know what is erroneously left in. I much prefer Chris's/Manda's almost nonsensical examples, because you can be sure this has nothing to do with the specific book, but they are good enough to explain what/how much/in what format we expect. 
* >Not everyone on our Book team has access to all the books, they can only access the books they're working on. (Unless I'm mistaken)

Every one who is "Team member" has access to all books in the online reader, even unpublished ones (once the books passes lint).
* >The reason I added longer sample texts in is because people just don't know what to write for certain parts of the book.

I don't know how common this is? I haven't seen a book that has missed out on team member stuff or back cover copy or anything, these have been all there. 
It is just knowing that some of this needs replicating in publish.yaml? (which is the thing that the guide should point out, maybe or the TODOs in publish.yaml to be changed to FPE: TODO).

* >We can't just link to examples as some parts don't show up online but they're important.

The only things that aren't shown are Bios and Dedication, and there is a plan to show them. I think it is enough to give them guidance how and what to write in the bio without examples (other than formatting hence Chris's "I like pie" dedication), most team members have had to write a bio for their raywenderlich.com user page — how is this any different? I also think that showing specific examples can take away from the personal aspect of both bios and dedications. Fine with the suggestions you've added about photos and like how to structure it, but maybe instead of an example it should just say "If you are unsure or struggling to write one, ask your book team for help/comments/examples")
The other thing that doesn't show is Back Cover Copy, but that gets replicated in Who's it for and Covered concepts, and possibly even Description. 

I think that FPEs are busy people, and adding lots of extra text to browse through rather than a concrete TODO is more likely to lead to the actual TODO being missed.
If you do think that specific examples are important then Sam's idea of a samples directory might be the best way forward, keeping the actual pages short and concrete. 
And sorry if there are many examples of these things that I am arguing against that I'm not aware of.

**Comment by rwenderlich** [comment]
Hi all, I just wanted to check in on this as it's been 9 days and we need this resolved for the Book FPE guide.

My understanding is everyone seems that everyone seems to be happy with the `samples` directory approach; that way we can provide an example in case people need it, but not bloat the file/instructions too much. If my understanding is correct, @filbabic could you set that up and @jellodiil @sammyd can then take another peek?

**Comment by filbabic** [comment]
@rwenderlich @jellodiil I've updated the PR according to our discussion (just pushed the changes, as I had some access issues yday).

Can you please take another small look and see if this is okay?

Thanks!

**Comment by filbabic** [comment]
@rwenderlich @jellodiil Just wanted to check in on this to see if there's anything left to change, or if we're clear to merge!

File: frontmatter/introduction.markdown [hunk 3]
```diff
@@ -20,16 +20,85 @@ This book is split into seventeen main sections:
...
 ### Section II: Section Name Here
 
-A paragraph or two about this section. Replicate exactly what you put in the section-intro-n.markdown file.
+A paragraph or two about this section. Replicate exactly what you put in the section-intro-2.markdown file.
 
 ### Section N: Section Name Here
 
 A paragraph or two about this section. Replicate exactly what you put in the section-intro-n.markdown file.
 
+# Example - Jetpack Compose Book
```
Comment (jellodiil) [inline frontmatter/introduction.markdown]: Hey, Is this meant to stay in the actual frontmatter? Can this go in the samples folder as well, please

**Review by jellodiil — CHANGES_REQUESTED** [review]
Looks good other than `frontmatter/introduction` still has the Jetpack Compose example

File: frontmatter/introduction.markdown (hunk 3 above)
Comment (filbabic) [reply frontmatter/introduction.markdown]: I'll move it to the samples! Sorry for missing that one!

**Comment by filbabic** [comment]
@jellodiil Removed the examples from the Introduction! :]

**Review by jellodiil — APPROVED** [review]
Looks good, thanks!

**Comment by filbabic** [comment]
@jellodiil I think you or @sammyd need to merge this as I'm a pleb with no auth rights! :]

**Pull request merged**